src/features/search/components/useHasPosition.ts

```diff
@@ -0,0 +1,13 @@
+import { LocationType } from 'features/search/enums'
+import { useSearch } from 'features/search/pages/SearchWrapper'
+import { useGeolocation } from 'libs/geolocation'
+
+export const useHasPosition = () => {
+  const { searchState } = useSearch()
+  const { position } = useGeolocation()
+
+  return (
+    searchState.locationFilter?.locationType !== LocationType.EVERYWHERE ||
+    (position !== null && searchState.locationFilter?.locationType === LocationType.EVERYWHERE)
+  )
+}
```

Comment:
Est-ce que tu peux ajouter un commentaire ou quelque chose stp ?
C'est assez compliqué de comprendre ce qu'il se passe.

Ou alors plutôt qu'un commentaire, extraire dans des variables plus explicites, pour que la personne qui passe derrière puisse comprendre ce qu'il se passe :P

Du genre
```suggestion
export const useHasPosition = () => {
  const { searchState } = useSearch()
  const { position } = useGeolocation()
  
  const searchEverywhere = searchState.locationFilter?.locationType === LocationType.EVERYWHERE
  const isUserLocated = position !== null
  
  return !searchEverywhere || (isUserLocated && searchEverywhere)
}
```

C'est un peu plus verbeux, mais plus facile à lire. Je ne sais pas ce qu'en pensent les autres devs mais j'aime bien faire comme ça.

Reply:
Tu as raison, un bon nommage suffit :)